src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2247,6 +2247,42 @@ internal void AddModuleInitializerMethod(MethodSymbol method)
             LazyInitializer.EnsureInitialized(ref _moduleInitializerMethods).Add(method);
         }
 
+        private ConcurrentSet<(InterceptsLocationAttributeData, MethodSymbol)>? _interceptions;
```

Comment:
>InterceptsLocationAttributeData [](http://example.com/codeflow?start=31&length=31)

I was surprised that we keep `AttributeLocation` in `InterceptsLocationAttributeData` in a `ConcurrentSet`. Should we instead keep only file-path, line and character number in that data type, as the key in a dictionary?
I understand there is follow-up to test with a large number of interceptors, so this can be addressed then.